src/Umbraco.Web.BackOffice/Filters/FileUploadCleanupFilterAttribute.cs

```diff
@@ -0,0 +1,163 @@
+    {
+        /// <summary>
+        ///     Constructor specifies if the filter should analyze the incoming or outgoing model
+        /// </summary>
+        /// <param name="incomingModel"></param>
+        public FileUploadCleanupFilterAttribute( bool incomingModel = true) : base(typeof(FileUploadCleanupFilter))
+        {
+
+            Arguments = new object[]
+            {
+                incomingModel
+            };
+        }
+
+        // We need to use IAsyncActionFilter even that we dont have any async because we need access to
+        // context.ActionArguments, and this is only available on ActionExecutingContext and not on
+        // ActionExecutedContext
+
+        private class FileUploadCleanupFilter : IAsyncActionFilter
+        {
+            private readonly ILogger _logger;
+            private readonly bool _incomingModel;
+
+            public FileUploadCleanupFilter(ILogger logger, bool incomingModel)
+            {
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _incomingModel = incomingModel;
+            }
+
+
+            public Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+            {
+                next(); // We only to do stuff after the action is executed
+
+                var tempFolders = new List<string>();
+
+                if (_incomingModel)
+                {
+                    if (context.ActionArguments.Any())
+                    {
+
+                        if (context.ActionArguments.First().Value is IHaveUploadedFiles contentItem)
+                        {
+                            //cleanup any files associated
+                            foreach (var f in contentItem.UploadedFiles)
+                            {
+                                //track all temp folders so we can remove old files afterwards
+                                var dir = Path.GetDirectoryName(f.TempFilePath);
+                                if (tempFolders.Contains(dir) == false)
+                                {
+                                    tempFolders.Add(dir);
+                                }
+
+                                try
+                                {
+                                    File.Delete(f.TempFilePath);
+                                }
+                                catch (Exception ex)
+                                {
+                                    _logger.Error<FileUploadCleanupFilterAttribute>(ex,
+                                        "Could not delete temp file {FileName}", f.TempFilePath);
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    if (context == null)
+                    {
+                        _logger.Warn<FileUploadCleanupFilterAttribute>("The context is null!!??");
```

Comment:
Very expressive warning message!

Reply:
Yeah, due to bugs in old mvc this used to throw because of bugs in async support in old net framework versions, was hard to track the bug down, these should never be thrown. Pretty sure thsi whole filter can be removed.

Reply:
The filter cannot be removed.. We still use it when we upload a file in one action and handle it in another action. Like import of doc types (next PR)